test/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigTest.java

```diff
@@ -66,27 +92,34 @@ public void canSerialize() throws Exception {
         appConfig.setModifiedOn(TIMESTAMP.getMillis());
         appConfig.setSurveyReferences(SURVEY_REFS);
         appConfig.setSchemaReferences(SCHEMA_REFS);
+        appConfig.setConfigReferences(CONFIG_REFS);
+        appConfig.setConfigElements(ImmutableMap.of("config1", TestUtils.getClientData()));
         appConfig.setClientData(clientData);
         appConfig.setVersion(3L);
         appConfig.setDeleted(true);
         
         Set<String> fields = Sets.newHashSet("criteria", "label", "createdOn", "modifiedOn", "clientData",
-                "surveyReferences", "schemaReferences", "version", "type", "deleted");
+                "surveyReferences", "schemaReferences", "configReferences", "configElements", "version", 
+                "type", "deleted");
                 
         JsonNode node = BridgeObjectMapper.get().valueToTree(appConfig);
         assertEquals(fields, Sets.newHashSet(node.fieldNames()));
         
+        assertEquals(clientDataString, node.get("clientData").toString());
         assertEquals("AppConfig", node.get("type").asText());
         assertEquals(TIMESTAMP.toString(), node.get("createdOn").textValue());
         assertEquals(TIMESTAMP.toString(), node.get("modifiedOn").textValue());
         
         AppConfig deser = BridgeObjectMapper.get().treeToValue(node, AppConfig.class);
         assertNull(deser.getStudyId());
+        assertEquals(clientDataString, deser.getClientData().toString());
+        assertEquals(clientDataString, deser.getConfigElements().get("config1").toString());
```

Comment:
I think there was a misunderstanding here. We don't want to verify the exact string conversion of the JSON, since that depends on a lot of factors (key ordering, formatting). Both here and in the JSON above.

However, we _do_ want to verify that the JSON contains the fields we expect in the format we expect (as opposed to some weird different format).

Reply:
Jackson serialization preserves order (unless you use @JsonPropertyOrder), and type is in the string (booleans, number and string values are different when serialized as JSON), but I can do this field by field if you want.